numpy/ma/tests/test_core.py

```diff
@@ -4732,6 +4732,49 @@ def test_ctor(self):
         assert_(not isinstance(m, np.ma.core.MaskedConstant))
         assert_(m is not np.ma.masked)
 
+    def test_repr(self):
+        # copies should not exist, but if they do, it should be obvious that
+        # something is wrong
+        assert_equal(repr(np.ma.masked), 'masked')
+        assert_not_equal(repr(np.ma.masked.copy()), 'masked')
+
+    def test_pickle(self):
+        from io import BytesIO
+        import pickle
+
+        with BytesIO() as f:
+            pickle.dump(np.ma.masked, f)
+            f.seek(0)
+            res = pickle.load(f)
+        assert_(res is np.ma.masked)
+
+    def test_write_to_copy(self):
+        # gh-9328
+        x = np.ma.masked.copy()
+        x[()] = 2
+
+        # write succeeds
+        assert_equal(x.data, 2)
+        assert_equal(x.mask, False)
+
+        # original should be unchanged
+        assert_equal(np.ma.masked.data, 0)
+        assert_equal(np.ma.masked.mask, True)
+
+    def test_no_copies(self):
+        MC = np.ma.core.MaskedConstant
+        assert_(not isinstance(np.ma.masked[...], MC))
+        assert_(not isinstance(np.ma.masked.copy(), MC))
```

Comment:
It is more that I had no idea what it would be! (As mentioned, I didn't look at the rest of the code, but from this PR it was not obvious what the result should be, and on current master it is a `MaskedConstant`)